examples/exex/rollup/src/db.rs

```diff
@@ -0,0 +1,280 @@
+use std::{
+    ops::Deref,
+    str::FromStr,
+    sync::{Arc, Mutex, MutexGuard},
+};
+
+use reth::revm::db::{
+    states::{PlainStorageChangeset, PlainStorageRevert},
+    BundleState,
+};
+use reth_primitives::{
+    revm_primitives::{AccountInfo, Bytecode},
+    Address, Bytes, SealedBlockWithSenders, B256, U256,
+};
+use reth_provider::{bundle_state::StorageRevertsIter, OriginalValuesKnown, ProviderError};
+use rusqlite::Connection;
+
+pub struct Database {
+    connection: Arc<Mutex<Connection>>,
+}
+
+impl Database {
+    /// Create new database with the provided connection.
+    pub fn new(connection: Connection) -> eyre::Result<Self> {
+        let database = Self { connection: Arc::new(Mutex::new(connection)) };
+        database.create_tables()?;
+        Ok(database)
+    }
+
+    fn connection(&self) -> MutexGuard<'_, Connection> {
+        self.connection.lock().expect("failed to acquire database lock")
+    }
+
+    fn create_tables(&self) -> eyre::Result<()> {
+        self.connection().execute_batch(
+            "CREATE TABLE IF NOT EXISTS block (
+                id     INTEGER PRIMARY KEY,
+                number TEXT UNIQUE,
+                data   TEXT
+            );
+            CREATE TABLE IF NOT EXISTS account (
+                id      INTEGER PRIMARY KEY,
+                address TEXT UNIQUE,
+                data    TEXT
+            );
+            CREATE TABLE IF NOT EXISTS account_revert (
+                id           INTEGER PRIMARY KEY,
+                block_number TEXT,
+                address      TEXT,
+                data         TEXT,
+                UNIQUE (block_number, address)
+            );
+            CREATE TABLE IF NOT EXISTS bytecode (
+                id   INTEGER PRIMARY KEY,
+                hash TEXT UNIQUE,
+                data TEXT
```

Comment:
kinda disappointed there is no native byte array type (`BLOB` is a diff story)

Reply:
everything is `TEXT` anyway 🫡

Reply:
hmm, I think we indeed should store arrays of uninterpretable bytes such as bytecodes as `BLOB`s. We don't need to compare them or see what's inside in a human-readable format, so no need for `TEXT` I believe.

Reply:
but it's a mega sqlite nit